src/common/linebuf-reader.cc

```diff
@@ -49,11 +50,25 @@ const char* LinebufReader::IncompleteLine() {
 
 const char* LinebufReader::ReadLine() {
   for (char *i = content_start_; i < content_end_; ++i) {
-    if (*i == '\n' || *i == '\r') {
-      *i = '\0';
-      const char *line = content_start_;
-      content_start_ = i + 1;
-      return line;
+    switch (ch_state_) {
+      case LF_SEEN: {
+        // Silently ignore newline
+        if (*i == '\n') {
```

Comment:
You are using the names opposite of the meaning of the characters here.
CR = Carriage Return = `\r`, and LF = Line Feed = `\n`.

Reply:
(solved, see below)